### PR TITLE
Domains: Fix contact edit screen usability

### PR DESCRIPTION
**Pull request by hambai**
We fixed some small issues with the domain contact screen usability.

#### Changes proposed in this Pull Request

* don't show the cancel button
* return to previous page on successful update
* remove unnecessary padding at the bottom

#### Testing instructions

* Open "Contacts and Privacy" and click on "Edit Contact Info"
* Verify that the "60 days transfer lock text" is readable
* Verify that there is no extra padding below the form button
* Verify that there is no Cancel button
* Verify that when contacts are successfully updated you're redirected back to the "Contacts and Privacy" screen

### CODE REVIEW

**Comment by fditrapani** [comment]
I just noticed the current message also includes "There may be a short delay before the changes show up in the public records." we should keep that as well.

**Comment by hambai** [comment]
@klimeryk I couldn't reproduce the button spacing issue... Is that still happening on your browser?

**Comment by klimeryk** [comment]
Yes, can still easily reproduce in `Firefox 72.0.2 (64-bit)`, macOS Catalina (10.15.2).

Doesn't seem to matter if the domain allows opting out of the 60 day lock or not:

Screenshot 2020-01-31 at 12 14 41
Screenshot 2020-01-31 at 12 15 01

I could not reproduce it in Chromium, so at least it's not something _completely_ weird 😅

**Comment by klimeryk** [comment]
Looks like if I remove `margin-bottom: 0` from `.contact-details-form-fields` it's fixed.

**Comment by hambai** [comment]
Yep, there was an issue with the css loading order and the css selectors priority - the `.contact-details-form-fields` selector was overriden by `.form-fieldset` when you loaded that from calypso.live with Firefox. Anyway - I've updated the css selector and now it should work fine.

File: client/components/domains/contact-details-form-fields/style.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.contact-details-form-fields {
+fieldset.contact-details-form-fields {
```
Comment (fditrapani) [inline client/components/domains/contact-details-form-fields/style.scss]: This change introduced a bug in the first name field layout. We need to set the top margin for that field to be zero.

**Comment by hambai** [comment]
Ugh... I'm not sure how I didn't see that - now it should be fixed.

There is another issue though I'm not sure how to fix - there is some extra spacing visible in Firefox that is not visible in Chrome - see the screenshots:

chrome
This is Chrome


firefox
And this is in Firefox

It seems that Chrome overlaps the margins of some of the elements, but I'm not sure how to handle that properly and would love some hints?

**Comment by hambai** [comment]
Oh - also the extra spacing issue is there even now, so it's not related to any of my changes

**Pull request merged**